test/suites/completions.sh

```diff
@@ -0,0 +1,330 @@
+    [ "$(complete list l)" = 'local:' ]
+    lxc remote switch local
+
+    # top level (instance) commands
+    lxc init testimage c1
+    lxc launch testimage c2
+    lxc snapshot c1
+    [ "$(complete console '')" = 'c1,c2,localhost:' ] # Console should show only instance server remotes and not the default
+    [ "$(complete console l)" = 'localhost:' ]
+    [ "$(complete console c)" = 'c1,c2' ]
+    [ "$(complete console localhost:c)" = 'localhost:c1,localhost:c2' ]
+    [ "$(complete copy '')" = 'c1,c2,localhost:' ] # Copy should show only instance server remotes and not the default
+    [ "$(complete copy l)" = 'localhost:' ]
+    [ "$(complete copy c)" = 'c1,c2' ]
+    [ "$(complete copy localhost:c)" = 'localhost:c1,localhost:c2' ]
+    [ "$(complete delete '')" = 'c1,localhost:' ] # Delete should show only instance server remotes and not the default, c2 should not be shown as it is not running
+    [ "$(complete delete l)" = 'localhost:' ]
+    [ "$(complete delete c)" = 'c1' ]
+    [ "$(complete delete --force '')" = 'c1,c2,localhost:' ] # Delete should show only instance server remotes and not the default, c2 should be shown as the --force flag was passed
+    [ "$(complete delete --force l)" = 'localhost:' ]
+    [ "$(complete delete --force c)" = 'c1,c2' ]
+    [ "$(complete delete --force localhost:c)" = 'localhost:c1,localhost:c2' ]
+    [ "$(complete exec '')" = 'c2,localhost:' ] # Exec should show only instance server remotes and not the default, c1 should not be shown because it is not running.
+    [ "$(complete exec l)" = 'localhost:' ]
+    [ "$(complete exec c)" = 'c2' ]
+    [ "$(complete exec localhost:c)" = 'localhost:c2' ]
+    [ "$(complete export '')" = 'c1,c2,localhost:' ] # Export should show only instance server remotes and not the default
+    [ "$(complete export l)" = 'localhost:' ]
+    [ "$(complete export c)" = 'c1,c2' ]
+
+    (
+      import_wd=$(mktemp -d -p "${TEST_DIR}" XXX)
+      # shellcheck disable=SC2164
+      # (tests are run with set -e)
+      cd "${import_wd}"
+      mkdir foo
+      touch bar.txt fizz.tar buzz.tar.gz bazz.tar.xz
+      [ "$(complete import '')" = 'buzz.tar.gz,foo/,localhost:' ] # Correct file extensions, directories, and non-default remotes
+      [ "$(complete import localhost: '')" = 'buzz.tar.gz,foo/' ]
+      rm -rf "${import_wd}"
+    )
+
+    [ "$(complete info '')" = 'c1,c2,localhost:' ] # Info should show only instance server remotes and not the default
+    [ "$(complete info l)" = 'localhost:' ]
+    [ "$(complete info c)" = 'c1,c2' ]
+    testimage_fingerprint="$(lxc query /1.0/images/aliases/testimage | jq -r .target)"
+    testimage_fingerprint_prefix="$(echo "${testimage_fingerprint}" | cut -c1-12)"
+    [ "$(complete init '')" = "${testimage_fingerprint_prefix},images:,localhost:,testimage,ubuntu-daily:,ubuntu-minimal-daily:,ubuntu-minimal:,ubuntu:" ] # Init should show images from the default remote, and all image remotes
```

Comment:
ATM, `lxc init` and `lxc launch` do not propose completing with fingerprints of local images. I believe the current behaviour is better as fingerprints are not so human friendly.

Having the alias (`testimage`) is good though but that's already the case now.

Reply:
Ok in that case I'll revert to only using aliases :)

Reply:
Do you think it's worth keeping the fingerprints for e.g. `image edit` completions? Or should we only ever offer completions for aliases? cc @tomponline

Reply:
I have updated the `cmpImages` function to only return image fingerprints when:
- there are no aliases for the image
- the image is not cached
- the remote is private

I think this is best of both, since users will never see fingerprint completions for public remotes (where aliases should be expected), nor will they see fingerprint completions when the image has a more friendly name elsewhere (e.g. they won't see the fingerprint of cached images from `images:`), but they will see their own images if they haven't created an alias for them yet.

Reply:
Sounds like a good heuristic to me, thanks!